category_encoders/hashing.py

```diff
@@ -238,18 +161,64 @@ def _transform_single_cpu(self, X, override_return_df=False):
         if not list(self.cols):
             return X
 
-        X = self.hashing_trick(X, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
-
-        if self.drop_invariant:
-            X = X.drop(columns=self.invariant_cols)
+        X = self.hashing_trick(
+            X, 
+            hashing_method=self.hash_method,
+            N=self.n_components, 
+            cols=self.cols, 
+        )
+        
+        return X
 
-        if self.return_df or override_return_df:
-            return X
-        else:
-            return X.to_numpy()
+    @staticmethod
+    def hash_chunk(shm_result, np_df, N, shm_offset):
+        for i, row in enumerate(np_df):
+            for val in row:
+                if val is not None:
+                    hasher = hashlib.md5()
+                    hasher.update(bytes(str(val), 'utf-8'))
+                    column_index = int.from_bytes(hasher.digest(), byteorder='big') % N
+                    row_index = (shm_offset + i)*N
+                    shm_index = row_index + column_index
+                    shm_result[shm_index] += 1
```

Comment:
how is this the same as the old code? wasn't the old code just doing `shm_result[column_index] += 1`?

Reply:
It was, but it was in the context of writting to a subset of the output dataframe. In that case we write to the "global" output dataframe, so we need to add the offset specific to the chunk we're processing.